dex/src/main/java/io/crate/data/join/AntiJoinNLBatchIterator.java

```diff
@@ -37,12 +38,12 @@
  *     }
  * </pre>
  */
-class AntiJoinBatchIterator<L, R, C> extends SemiJoinBatchIterator<L, R, C> {
+class AntiJoinNLBatchIterator<L extends Row, R extends Row, C extends Row> extends SemiJoinNLBatchIterator<L, R, C> {
```

Comment:
Why do we introduce the Row constraint? The reason I added the `ElementCombiner` was to keep the BatchIterator decoupled from any specific item types. Maybe this was over-engineered because we never use it for anything but `Row`. Either way, if we now introduce this constraint I think we should also remove the `ElementCombiner`.

Reply:
@mfussenegger In the new HashJoinBatchIterator I need to materialize so I need to cast the to (Row) and then materialize. Do you think it's ok to go ahead and make it strict for Row only or keep it as it was?

Reply:
Wouldn't it be sufficient in that case to have the `Row` type constraint on the `HashJoinBatchIterator` in that case but leave the existing ones as they are?

Reply:
Somehow I thought I couldn't "shadow" that